src/encoder.rs

```diff
@@ -90,6 +91,7 @@ impl fmt::Display for FormatError {
             EndReached => write!(fmt, "all the frames have been already written"),
             MissingFrames => write!(fmt, "there are still frames to be written"),
             MissingData(n) => write!(fmt, "there are still {} bytes to be written", n),
+            Unrecoverable => write!(fmt, "the writer is in an unrecoverable state"),
```

Comment:
Isn't the error more that the writer was _used_ after a previous operation put the writer into an unrecoverable state?

Reply:
So something along the lines of:
> a previous error put the writer into an unrecoverable state